cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/config/WebSecurity.java

```diff
@@ -97,4 +318,94 @@ ReactiveUserDetailsService userDetailsService() {
         };
     }
 
+    static String getClientRegistrationId(ServerWebExchange exchange) {
+        var path = exchange.getRequest().getPath().value();
+        var matcher = CLIENT_REG_ID.matcher(path);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        }
+        throw new IllegalStateException("Client registration ID was not found in the path: " + path);
```

Comment:
Is `IllegalStateException` the proper exception to throw in case of missing client ID?

Reply:
this exception is probably proper as the state should not happen